packages/fuel-gauge/src/contract-factory.test.ts

```diff
@@ -245,21 +253,232 @@ describe('Contract Factory', () => {
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const deploy = await ConfigurableContractFactory.deploy(wallet);
+    const { contract } = await deploy.waitForResult();
+
+    const call = await contract.functions.echo_u8().call();
+    const { value } = await call.waitForResult();
+    expect(value).toBe(10);
+  });
+
+  it('deploys a large contract via deploy entrypoint', async () => {
+    using launched = await launchTestNode({
+      providerOptions: {
+        cacheUtxo: -1,
+      },
+    });
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const deploy = await LargeContractFactory.deploy(wallet);
+    const { contract } = await deploy.waitForResult();
+
+    const call = await contract.functions.something().call();
+    const { value } = await call.waitForResult();
+    expect(value.toNumber()).toBe(1001);
+  }, 15000);
+
+  it('should not deploy large contract with invalid balance', async () => {
+    using launched = await launchTestNode({
+      providerOptions: {
+        cacheUtxo: -1,
+      },
+      walletsConfig: {
+        amountPerCoin: 0,
+      },
+    });
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const factory = new ContractFactory(LargeContractFactory.bytecode, LargeContract.abi, wallet);
+
+    await expectToThrowFuelError(
+      () => factory.deploy({ maxFee: 0 }),
+      new FuelError(ErrorCode.FUNDS_TOO_LOW, 'Insufficient balance to deploy contract.')
+    );
+  });
+
+  it('deploys a contract via blobs [configurables]', async () => {
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const factory = new ContractFactory(
+      ConfigurableContractFactory.bytecode,
+      ConfigurableContract.abi,
+      wallet
+    );
+
+    const deploy = await factory.deployContractAsBlobs({
+      configurableConstants: {
+        U8: 1,
+        U16: 2,
+        U32: 3,
+        U64: 4,
+      },
+    });
+    const { contract } = await deploy.waitForResult();
+
+    const u8Call = await contract.functions.echo_u8().call();
+    const u8Result = await u8Call.waitForResult();
+    expect(u8Result.value).toBe(1);
+
+    const u16Call = await contract.functions.echo_u16().call();
+    const u16Result = await u16Call.waitForResult();
+    expect(u16Result.value).toBe(2);
+
+    const u32Call = await contract.functions.echo_u32().call();
+    const u32Result = await u32Call.waitForResult();
+    expect(u32Result.value).toBe(3);
+
+    const u64Call = await contract.functions.echo_u64().call();
+    const u64Result = await u64Call.waitForResult();
+    expect(u64Result.value.toNumber()).toBe(4);
+  }, 15000);
+
+  it('deploys a contract via blobs [storage]', async () => {
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const factory = new ContractFactory(
+      StorageTestContractFactory.bytecode,
+      StorageTestContract.abi,
```

Comment:
Do we want to be testing that the `LargeContract` can be deployed with storage?

Reply:
As [above](https://github.com/FuelLabs/fuels-ts/pull/2827#discussion_r1713534682), the only difference is the number of blobs. This ensures that storage persists when deploying as blobs.